src/block/stdio_executor.rs

```diff
@@ -25,30 +25,53 @@
 
 use std::fmt::{self, Display};
 use std::io::{Read, Seek, SeekFrom, Write};
-use std::{io, result};
+use std::{io, mem, result};
 
-use vm_memory::{Bytes, GuestMemory, GuestMemoryError};
+use vm_memory::{ByteValued, Bytes, GuestMemory, GuestMemoryError};
 use vmm_sys_util::file_traits::FileSync;
-use vmm_sys_util::write_zeroes::{PunchHole, WriteZeroes};
+use vmm_sys_util::write_zeroes::{PunchHole, WriteZeroesAt};
 
 use crate::block::{
-    defines::{SECTOR_SHIFT, SECTOR_SIZE, VIRTIO_BLK_F_FLUSH, VIRTIO_BLK_F_RO, VIRTIO_BLK_T_FLUSH},
+    defines::{
+        SECTOR_SHIFT, SECTOR_SIZE, VIRTIO_BLK_F_DISCARD, VIRTIO_BLK_F_FLUSH, VIRTIO_BLK_F_RO,
+        VIRTIO_BLK_F_WRITE_ZEROES, VIRTIO_BLK_T_DISCARD, VIRTIO_BLK_T_FLUSH,
+        VIRTIO_BLK_T_WRITE_ZEROES, VIRTIO_BLK_WRITE_ZEROES_FLAG_UNMAP,
+    },
     request::{Request, RequestType},
 };
 
 /// Trait that keeps as supertraits the ones that are necessary for the `StdIoBackend` abstraction
 /// used for the virtio block request execution.
-pub trait Backend: Read + Write + Seek + FileSync + PunchHole + WriteZeroes {}
+pub trait Backend: Read + Write + Seek + FileSync + PunchHole + WriteZeroesAt {}
+
+impl<B: Read + Write + Seek + FileSync + PunchHole + WriteZeroesAt> Backend for B {}
+
+/// One or more `DiscardWriteZeroesData` structs are used to describe the data for
+/// discard or write zeroes command.
+#[derive(Copy, Clone, Debug, Default)]
+#[repr(C)]
+struct DiscardWriteZeroesData {
```

Comment:
nit: we can drop `Data` from the name so that it maps to the virtio spec name better.

Suggestion: how about we move the definition of `VIRTIO_BLK_WRITE_ZEROES_FLAG_UNMAP` to the implementation of this structure? This way we can shorten the name quite a bit and just call it `UNMAP_BIT`, and the definition is also put in context.

Reply:
done